lib/plugins/aws/package/compile/events/apiGateway/lib/requestValidator.js

```diff
@@ -0,0 +1,209 @@
+'use strict';
+
+const _ = require('lodash');
+const BbPromise = require('bluebird');
+
+module.exports = {
+  compileRequestValidators() {
+    const apiGateway = this.serverless.service.provider.apiGateway || {};
+    const models = {};
+    // Models are defined in the provider and can be used in http events later
+    if (apiGateway.requestSchemas) {
+      for (const [schemaId, schemaConfig] of _.entries(apiGateway.requestSchemas)) {
+        let modelName;
+        let validatorName;
+        let description;
+        let definition;
+
+        // If schema is not defined this will try to map resourceDefinition as the schema
+        if (!schemaConfig.schema) {
+          definition = schemaConfig;
+        } else {
+          definition = schemaConfig.schema;
+        }
+
+        const modelLogicalId = this.provider.naming.getModelLogicalId(schemaId);
+
+        const validatorLogicalId = this.provider.naming.getValidatorLogicalId(modelLogicalId);
+
+        if (schemaConfig.name) {
```

Comment:
To be on a safe side, I believe this condition should be run only if `definition` was provided on `schemaConfig.schema` property.

Otherwise we're inspecting the actual _definition_ for those properties, and e.g. `description` is a perfectly valid JSON Schema property, which with logic as proposed may be used as model description, which is not right